hnix-store-remote/src/System/Nix/Store/Remote/Protocol.hs

```diff
@@ -168,17 +171,25 @@ runStore = runStoreOpts defaultSockPath "/nix/store"
 
 runStoreOpts
   :: FilePath -> FilePath -> MonadStore a -> IO (Either String a, [Logger])
-runStoreOpts sockPath storeRootDir code = do
-  bracket (open sockPath) (Network.Socket.close . storeSocket) run
+runStoreOpts path = runStoreOptsInner S.AF_UNIX (SockAddrUnix path)
+
+runStoreOptsTCP
+  :: String -> Int -> FilePath -> MonadStore a -> IO (Either String a, [Logger])
+runStoreOptsTCP host port storeRootDir code = do
+  let hints = S.defaultHints
+  S.getAddrInfo (Just hints) (Just host) (Just $ show port) >>= \case
+    (sockAddr:_) -> runStoreOptsInner (S.addrFamily sockAddr) (S.addrAddress sockAddr) storeRootDir code
+    _ -> return (Left "Couldn't resolve host and port with getAddrInfo.", [])
```

Comment:
`return` entered deprecation in GHC 9.2 https://github.com/fumieval/Haskell-breaking-changes#planned-remove-return-from-monad.